### DISCUSSION TITLE
Features do not toggle when activated outside of PC?

### DISCUSSION THREAD

**Discussion by calisro** [rstrouse/nodejs-poolController-dashPanel#80] [Q&A]
I currently have a screenlogic2 (SL) and also poolcontroller (PC) connected via RS485 daisy chained for testing.  i use Home Assistant (HA) currently with its default integration to control screenlogic (SL).  My goal is to do away with SL as I desire better control over the pool equipment and I do not see a need for the SL device as it is only a control point for HA integration.  

(Disclaimer, I did not program the EasyTouch 4).

That being said, I have been trying to test PC when I perform actions through SL (via HA).  Whenever I turn a feature on outside of the PC Dash, that feature doesn't show 'on/off' in the PC Dash.  When I action the feature through SL/HA though, it is correctly reflected in both Screenlogic/HA and PC.  

So for example, I have a feature for 'pool' and 'pool high'  

If I turn pool on in SL/HA, pool feature does not show on in PC.  I only see the pump turn on.
If I turn the feature pool on in PC, I see it show on in SL/HA correctly.  

Is this expected behavior?  Do I have something wrong?  Is it a bug?  

![image](https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/f486cd52-2e73-44f4-b4a1-363634c4c534)

**Comment by tagyoureit** [comment]
Can you please be a little more specific about your installation and which packages you have connected to which apps?

**Reply by calisro** [reply]
Hi.  Sure.

I have a Screenlogic2 connected to a Easy touch2.  In series with the Screenlogic2 I also have an RS485 with nodejs-PoolController running on a Raspberry pi along with the PoolController-Dash.  To be clear:  I do not have PoolController connected to the screenlogic.  They are connected separately in series to the EasyTouch.

The Screenlogic device is integrated to [Home Assistant Pentair integration](https://www.home-assistant.io/integrations/screenlogic/) for which I have quite a lot of automations setup for (This is my original integration which currently runs everything).  I also have but rarely use the slconfig android app from Pentair.

The nodejs-PoolController is integrated to the PoolController-Dash *and* also the [njsPC-HA](https://github.com/Crewski/njsPC-HA).  


Essentially this is it.  Hopefully its readable.

![image](https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/95c654a6-3753-49f0-a4de-4c620482cae5)



In the screenshot above you're seeing the difference if I toggle the 'pool' on with an app intergated to screenlogic versus if I toggle it with the PoolController-Dash or even njsPC-HA.  

If I toggle it with njsPC-HA or PoolController-Dash = screenlogic connected devices know it was toggled and correctly reflect the change.  

If I toggle it with screenlogic connected integration or slconfig app = Dash and njsPC-HA both have no idea the feature circuit was changed.  THey do show the intelliflow RPMS/wattages/pump itself on only.

**Reply by tagyoureit** [reply]
The HomeAssistant-Pentair integration looks like it only changes the light colors.  Screenlogic tries to keep track of that but there isn't anything on ET that would know about the color changes.  

Inside njsPC, we keep track of which color is selected, but if Screenlogic changes the color we have no way to know about it.

In short, if you can use njsPC-HA do you still need the HomeAssistant-Pentair integration?  Most users who have ScreenLogic don't use it at all when they can use either dashPanel or HomeAssistant directly.

**Reply by calisro** [reply]
Oh this isn't a light issue at all.  I can certainly understand not knowing what light color was selected.   Easytouch itself doesn't even know that.

It's the pumps, blowers, waterfalls, etc,etc. Anything that is done outside of njs-Dash is completely blind within njsDash.  If I walk outside and turn the pump, spa, spa high, waterfall, blower, etc on from the easytouch control panel, njs-Dash has no idea it's happened.  Screenlogic does though.  That is a problem. 

> In short, if you can use njsPC-HA do you still need the HomeAssistant-Pentair integration? Most users who have ScreenLogic don't use it at all when they can use either dashPanel or HomeAssistant directly.

That's really my dilemma.  I have a large amount of automation, sensors, history trackers, buttons, dashboards, etc that are all currently hooked up through ScreenLogic-HA and while I'd like to wholy move away from SL, I would like to transition over and verify that things are working as expected.  

What really bothers me though is:
- Any device I turn on/off through NJS: screenlogic knows and picks it up.  
- Most devices I turn on/off through the easytouch directly or through Screenlogic: njs is completely blind. 

I created this video of a dashboard which demonstrates the issue.  Keep in mind that ALL of these same functions if I do from the Easytouch controlpanel itself, njs-Dash acts the same way!  blind.  THe devices that are prefixed with PC are njs-PC (But njs-dash shows the same blindness) and devices that are prefixed SL are Screenlogic controlled (but same happens at the easy touch)

https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/95f9e192-7612-4a5a-ba2f-b4d08940285e

The air/water temps are all wrong as well since they haven't updated. :)  Screenlogic ones are correct.

**Reply by tagyoureit** [reply]
Can you check the circuit changes against njsPC dashPanel?  @crewski maintains the HA integration but the first place to look would be in dashPanel.

**Reply by calisro** [reply]
As you can see in this video, It is exactly the same in the Dash. If anything is toggled at the EasyTouch Panel or Screenlogic, Dash is blind.  If they are toggled in the Dash, the Easytouch and screenlogic are aware.

The interesting thing is we can see the broadcast messages in the njs-dash message manager but they are not being listened to.  I assume because of the source. It seems the Dash/HA plugin listen for anything originating to/from the plugin/dash only.  

https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/72856106-23c2-49df-864b-957aece62f5d

![276395490-f486cd52-2e73-44f4-b4a1-363634c4c534](https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/5b3a30c0-a0d4-423f-a034-438bd86389a1)

**Reply by calisro** [reply]
Here's another with all 3 windows at once (message manager with some filtering, NJS-Dash, SL/njs-ha via HA).


https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/b8753172-ea16-4e34-834a-3b270f987162

**Reply by tagyoureit** [reply]
We don't update the status of a circuit based on the request (or response) for that circuit on the message bus.  We only update the status when the 2 packet (equipment status packet) is broadcast.  Your ET isn't broadcasting that, for some reason.  Or do you have filters on to only show certain packets?

**Reply by calisro** [reply]
hm.  I see. I do have filtering.  This is a quick non-filtered.  Its so many messages so I was trying to cut it down. 

> We only update the status when the 2 packet (equipment status packet) is broadcast. 


I am not sure what to look for here.  but I did another capture and video.  I'm not sure how to download that capture if that would help.

https://github.com/rstrouse/nodejs-poolController-dashPanel/assets/5018276/ea146b56-119d-4e21-bd4b-61feaa77e548
